### PR TITLE
i18n(es): Update astro-components

### PR DESCRIPTION
**Pull request by glopzel**
update info on dynamic variables

#### What kind of changes does this PR include?

- New or updated content
- Translated content

#### Description

- What does this PR change? There used to be a note for specifications on dynamic variables, now there's more of a detailed explanation.

### CODE REVIEW

File: src/pages/es/core-concepts/astro-components.md [hunk 1]
```diff
@@ -183,9 +183,11 @@ const Component = MyComponent;
 <Component /> <!-- es renderizado como <MyComponent /> -->
 ```
 
-:::note
-Los nombres de las variables deben estar en mayúscula (`Element`, no `element`), para que esto funcione. De otra manera, Astro intentará renderizar el nombre de tu variable como una etiqueta literal de HTML.
-:::
+Al usar etiquetas dinámicas:
+
+**Nombres de las variables deben  estar en mayúscula**. Por ejemplo, usar `Element`, no `element`.  De lo contrario, Astro intentará renderizar el nombre de la variable como una etiqueta literal de HTML.
```
Comment (agustinmulet) [inline src/pages/es/core-concepts/astro-components.md:188]:
```suggestion
**Los nombres de las variables deben estar en mayúscula**. Por ejemplo, usar `Element`, no `element`. De lo contrario, Astro intentará renderizar el nombre de la variable como una etiqueta literal de HTML.
```

File: src/pages/es/core-concepts/astro-components.md [hunk 2]
```diff
@@ -183,9 +183,11 @@ const Component = MyComponent;
...
+
+**Directivas de hidratación no tienen soporte**. Al usar [`client:*` directiva de hidratación](/es/core-concepts/framework-components/#hidratando-componentes-interactivos), Astro necesita saber cuales son los componentes que se deben empaquetar para producción, y el patrón de etiqueta dinámica previene que esto funcione.
```
Comment (agustinmulet) [inline src/pages/es/core-concepts/astro-components.md:190]:
```suggestion
**No admiten directivas de hidratación**. Al usar [`client:*` directiva de hidratación](/es/core-concepts/framework-components/#hidratando-componentes-interactivos), Astro necesita saber cuáles son los componentes que se deben empaquetar para producción, y el patrón de etiqueta dinámica previene que esto funcione.
```

**Review by agustinmulet — COMMENTED** [review]
Revisado con un par de sugerencias, buen trabajo! 💪

**Review by agustinmulet — APPROVED** [review]
LGTM! ✅

**Review by delucis — APPROVED** [review]
Woah — just updated these docs and already translated 😮 

Awesome work team 💜

**Pull request merged**